docs/design-documents/tools/cmake_components.md

```diff
@@ -45,9 +45,32 @@ targets:
 
 Breaking the dependencies would be a huge effort and it possibly would result in rewriting these components as they were for years considered as monolitic - having everything available.
 
-## Components as object libraries
+## Components libraries
 
-CMake provides OBJECT libraries but it does not support circular dependencies that we have in our tree. Therefore we build Mbed OS as whole (all object files combined).
+Mbed OS uses various type of CMake libraries:
+
+`STATIC` libraries should be the default library type used to build Mbed OS CMake
+libraries. However, due to its inability to resolve weakly linked
+symbols with strong ones, we have to use other types when a library
+contains weak symbols.
+
+`OBJECT` libraries are used in Mbed OS to generates objects file but not archiving
+them. It has the advantage of not always rebuilding files if there aren't any
+changes and can resolve weakly linked symbols with strong ones. Its main
+disadvantage is that it does not allow circular dependency between libraries.
```

Comment:
Can we list the main two problems we hit with OBJECT:
https://gitlab.kitware.com/cmake/cmake/-/issues/18090
https://gitlab.kitware.com/cmake/cmake/-/issues/17905

These 2 are the main that we could reproduce with Mbed OS codebase

Reply:
Isn't https://gitlab.kitware.com/cmake/cmake/-/issues/18090 what I mentioned in the note section?
And https://gitlab.kitware.com/cmake/cmake/-/issues/17905 is already mentioned here.
Do you mean listing the actual Gitlab issues?

Reply:
[Review](https://github.com/ARMmbed/mbed-os/compare/7d56c96f6e67561b8f1a825b5780d8cb8178b3a0..b4721d1c8a9b5becb69e93d60a97c075f67fafa4).